### ISSUE TITLE
GameGuru MAX - Request - BSP-Like Editor to Replace EBE?

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [TheGameCreators/GameGuruRepo#3455]
Okay, this one is WAAAAAAY out there and probably would never get implemented, but I think it would be a pretty cool addition to MAX. Having said that, let's dive in anyway. ;)

EBE is an interesting idea and it's attractive for people who can't create their own models because it allows them to create structures and easily apply materials to walls, floors, ceilings, etc. However, EBE is very limited because all walls are straight and can only be placed at right angles, etc. 

The other day I was playing with a Quake editor called TrenchBroom:

https://trenchbroom.github.io/

It reminded me of the days when I used to edit old BSP maps, creating brushes and easily applying textures that (usually) just worked as they were supposed to. I thought about how easy it was to create interior maps with these editors. Then I recalled having worked with an engine called Leadwerks and how it's editor was BSP like, even if the engine wasn't a BSP engine. And all of that got me thinking ...

**_... Wouldn't it be cool if MAX had a BSP-like interior level editor instead of EBE? It would allow for a lot more flexibility and allow for the end-user to more easily apply textures/materials as they see fit._**

I realize this would be a lot of work, but it would be cool, I think. And it would be fun to use (if developed right). So, yeah, this is way out there, but it's a request so ... ;)

**Comment by MonkeyFrogStudio** [comment]
One of the things I remember about using these old BSP/Brush editors is that it was how easy it was to apply textures to your surfaces. Having something like this in MAX would open a lot of doors for people who can't or don't want to use a modeling program to create interior levels. No UV mapping is needed as that's handled via the brushes (though many of these apps had tools to adjust even that). A tool like this in MAX (or made for use with MAX) would really allow non-3D artist to create awesome interior levels.

Not only that, but it would facilitate the ability to create "old-school" looking shooters similar to DOOM, Quake, etc. (at least in style of architecture). These types of games are all-the-rage currently (for those that like FPS games).

As a result of ease-of-use and the ability to simulate the look of old-school games (but with PBR materials!), I think this would be a great selling point for MAX, too.

**Comment by earthling45** [comment]
https://www.youtube.com/watch?v=tRpUagKAqvA

A short video so people have a better understanding of what you mean with BSP. :)

I actually do think that EBE is already quite geared towards such a builder, it only misses the shapes or should i say brushes?.
They could of course be added and if the FPE can be edited within Max, it should be easy to apply different textures.

Back in 2017 i did create a tutorial for EBE, it is amazing what can be achieved with a few blocks and a bit of math.

**Comment by earthling45** [comment]
Just thought of this, now that we are able to group entities i should be able to build that church from the tutorial again, save it as a grouped entity and drag it in any level i create. :)

**Comment by MonkeyFrogStudio** [comment]
The problem with EBE is that you add a wall and, well, that's it. You're stuck with that wall at that size and thickness. The EBE grid is what you have to work with for laying down floor tiles, walls, ceiling, etc. With something akin to BSP brushes, you would not be restricted in such a manner. Clever designers could also develop caverns, if they wanted to; "curved" walls and corridors; odd-shaped doors, and more. It would be easy for most people to do the basics (very similar to EBE) while not restricting the more adventurous.

**Comment by earthling45** [comment]
True, but how are you going to handle the texture tiling on curved walls?
It was on my list to make prefabs for EBE so that we could have and use arched doorways and windows but what put me of was the texel density on EBE, to much loss of detail.

**Comment by MonkeyFrogStudio** [comment]
"Curved" walls have been simulated in BSP for decades. And tiled textures work fine on them. That's why these programs, like Trenchbroom (and even the old Quark) had a simple means to adjust textures to ensure they fit/tiled/could be rotated, etc.

**Comment by davetgc** [comment]
There are plans for an interior level editor. V1 certainly won't be of this level, but will enable users to create decent levels quickly.
